flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/delegation/FlinkSqlParserImplFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.delegation;
+
+import org.apache.flink.sql.parser.hive.impl.FlinkHiveSqlParserImpl;
+import org.apache.flink.sql.parser.impl.FlinkSqlParserImpl;
+import org.apache.flink.sql.parser.validate.FlinkSqlConformance;
+
+import org.apache.calcite.sql.parser.SqlAbstractParserImpl;
+import org.apache.calcite.sql.parser.SqlParserImplFactory;
+import org.apache.calcite.sql.validate.SqlConformance;
+
+import java.io.Reader;
+
+/**
+ * A SqlParserImplFactory that creates the parser according to SqlConformance.
+ */
+public class FlinkSqlParserImplFactory implements SqlParserImplFactory {
+
+	private final SqlConformance conformance;
+
+	public FlinkSqlParserImplFactory(SqlConformance conformance) {
+		this.conformance = conformance;
+	}
+
+	@Override
+	public SqlAbstractParserImpl getParser(Reader stream) {
+		if (conformance == FlinkSqlConformance.HIVE) {
+			return FlinkHiveSqlParserImpl.FACTORY.getParser(stream);
+		} else {
+			return FlinkSqlParserImpl.FACTORY.getParser(stream);
+		}
+	}
```

Comment:
We do not need to extend `SqlParserImplFactory `, how about just a tool class named `FlinkSqlParserFactories` and there is a method to return the factory by conformance `FlinkSqlParserFactories#createFactory(SqlConformance)` ?

Reply:
OK, sounds good.